src/api/routes/offer.js

```diff
@@ -28,6 +28,38 @@ module.exports = (app) => {
         }
     });
 
+    /**
+     * Gets an offer from the database with its id
+    */
+    router.get("/:offer", validators.get, async (req, res, next) => {
+        try {
+            const offer = await (new OfferService()).getOfferByID({ id: req.params.offer });
```

Comment:
Since the getOfferById only receives an id, the argument should be the id and not an object with an id.

Also, rename to `getOfferById` (lowercase i) to make it consistent with other `*byId` methods